File: 136.只出现一次的数字.py
#
# @lc app=leetcode.cn id=136 lang=python3
#
# [136] 只出现一次的数字
#

# @lc code=start
from typing import Dict, List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        res = 0
        for num in nums:
            res ^= num
        return res


# @lc code=end

File: 136.只出现一次的数字.ts
/*
 * @lc app=leetcode.cn id=136 lang=typescript
 *
 * [136] 只出现一次的数字
 */

// @lc code=start
function singleNumber(nums: number[]): number {
    if (nums.length === 1) {
        return nums[0];
    }

    let res = 0;
    for (const num of nums) {
        res ^= num;
    }
    return res;
}

// @lc code=end
